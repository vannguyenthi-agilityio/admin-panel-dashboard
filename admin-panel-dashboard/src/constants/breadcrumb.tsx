// Types
import { IItemBreadcrumb } from "@/types";

// Constants
import { ROUTES } from "@/constants";

export const ITEM_BREADCRUMB: IItemBreadcrumb[] = [
  {
    href: ROUTES.HOME,
    crumbName: "DashBoard",
    linkActiveCrumb: false,
    linkIndex: 1
  },
  {
    href: ROUTES.CUSTOMERS,
    crumbName: "Customersâ€™ List",
    linkActiveCrumb: true,
    linkIndex: 2
  }
]

export const ITEM_BREADCRUMB_FORM: IItemBreadcrumb[] = [
  {
    href: ROUTES.HOME,
    crumbName: "DashBoard",
    linkActiveCrumb: false,
    linkIndex: 1
  },
  {
    href: ROUTES.CUSTOMERS,
    crumbName: "Customersâ€™ List",
    linkActiveCrumb: false,
    linkIndex: 2
  },
  {
    href: `${ROUTES.CUSTOMERS}/form`,
    crumbName: "New Customer",
    linkActiveCrumb: true,
    linkIndex: 3
  }
]

export const ICON_ITEM_BREADCRUMB: IItemBreadcrumb[] = [
  {
    href: ROUTES.HOME,
    crumbName: "DashBoard",
    linkActiveCrumb: false,
    linkIndex: 1,
    iconElement: <span className="mr-1">ðŸ‘‰</span>,
  },
  {
    href: ROUTES.CUSTOMERS,
    crumbName: "Customersâ€™ List",
    linkActiveCrumb: true,
    linkIndex: 2,
    iconElement: <span className="mr-1">ðŸ‘‰</span>,
  }
]
