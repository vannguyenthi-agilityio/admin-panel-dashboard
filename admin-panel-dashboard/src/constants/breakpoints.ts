export const BREAKPOINTS = {
  MD: 768,
  XL: 1280,
};
