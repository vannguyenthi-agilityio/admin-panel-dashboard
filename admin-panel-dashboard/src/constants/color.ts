export const COLOR = {
  PRIMARY: {
    50: '#fff'
  },
  SECONDARY: {
    50: '#000',
    100: '#adb5bd',
    200: '#212529'
  },
}
