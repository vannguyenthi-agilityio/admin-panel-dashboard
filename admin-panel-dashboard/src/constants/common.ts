// Types
import { OptionType } from "@/types";

export const MONTHS: OptionType[] = [
  {
    option: "January",
    value: "0",
  },
  {
    option: "February",
    value: "1",
  },
  {
    option: "March",
    value: "2",
  },
  {
    option: "April",
    value: "3",
  },
  {
    option: "May",
    value: "4",
  },
  {
    option: "June",
    value: "5",
  },
  {
    option: "July",
    value: "6",
  },
  {
    option: "August",
    value: "7",
  },
  {
    option: "September",
    value: "8",
  },
  {
    option: "October",
    value: "9",
  },
  {
    option: "November",
    value: "10",
  },
  {
    option: "December",
    value: "11",
  },
];
