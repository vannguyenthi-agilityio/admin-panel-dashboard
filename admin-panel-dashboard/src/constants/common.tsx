// Types
import { OptionType } from "@/types";

export const MONTHS: OptionType[] = [
  {
    option: "January",
    value: "0",
  },
  {
    option: "February",
    value: "1",
  },
  {
    option: "March",
    value: "2",
  },
  {
    option: "April",
    value: "3",
  },
  {
    option: "May",
    value: "4",
  },
  {
    option: "June",
    value: "5",
  },
  {
    option: "July",
    value: "6",
  },
  {
    option: "August",
    value: "7",
  },
  {
    option: "September",
    value: "8",
  },
  {
    option: "October",
    value: "9",
  },
  {
    option: "November",
    value: "10",
  },
  {
    option: "December",
    value: "11",
  },
];

export const PAGINATION : OptionType[] = [
  {
    option: 1,
    value: "1",
  },
  {
    option: 2,
    value: "2",
  },
  {
    option: 3,
    value: "3",
  },
  {
    option: 4,
    value: "4",
  },
  {
    option: 5,
    value: "5",
  },
  {
    option: 6,
    value: "6",
  },
  {
    option: 7,
    value: "7",
  },
  {
    option: 8,
    value: "8",
  },
  {
    option: 9,
    value: "9",
  },
  {
    option: 10,
    value: "10",
  },
  {
    option: 11,
    value: "11",
  },
]

export const RESULT_NOT_FOUND = "Result Not Found";

export const WINDOW_INNER_WIDTH = window.innerWidth;

export const PREFIX_PHONE_NUMBER = "+44";

export const FORMAT_DATE = {
  YEAR_MONTH_DATE_TIME: "YYYY-MM-DDTHH:mm:ss.SSS",
  MONTH_DAY_YEAR_TIME: "MMM D, YYYY h:mm A",
  MONTH_DAY_YEAR: "MM/DD/YYYY"
};

export enum ACTION_TYPE {
  EDIT = 'Edit',
  CREATE = 'Create',
  DETAIL = 'Detail',
  DELETE = 'Delete'
}

export enum FORM_TYPE {
  EDIT = 'Edit',
  CREATE = 'Create',
  DETAIL = 'Detail'
}

export enum SCREEN_SIZE {
  XS = "xs",
  SM = "sm",
  MD = "md",
  LG = "lg",
  XL = "xl"
}
