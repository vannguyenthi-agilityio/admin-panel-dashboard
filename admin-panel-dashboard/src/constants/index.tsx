export * from './color';
export * from "./common";
export * from "./routes";
export * from "./breadcrumb";
export * from "./message";
export * from "./routes";
export * from "./breakpoints";
export * from "./regex";
