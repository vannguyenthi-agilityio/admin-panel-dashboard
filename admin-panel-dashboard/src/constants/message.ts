export const MESSAGES_ERROR = {
  FIELD_REQUIRED: "Field is required.",
  NAME_REQUIRED: "Name is required.",
  EMAIL_REQUIRED: "Email is required.",
  EMAIL_INVALID: "Invalid email address.",
  MIN_LENGTH_2: "This field must have more than 2 characters.",
  NAME_INVALID: "This field is no space at start or end",
  ID_NUMBER_INVALID: "Invalid number.",
  INVALID_PHONE_NUMBER: "Phone number is invalid",
  INVALID_BIRTH_DATE: "Birth date is invalid.",
  INVALID_PASSWORD:
    "Password must have at least 8 characters, 1 special character and 1 number.",
  UNKNOWN_ERROR: 'An unknown error occurred',
  CUSTOMER_CONTEXT_ERROR: 'useActionData hooks should using inside CustomerProvider!',
  TOAST_CONTEXT_ERROR: 'useToast should called in ToastProvider!',
};

export const MESSAGES_WARNING = {
  ID_NUMBER: "The ID number should be 10 characters"
}

export const MESSAGE_ADD_CUSTOMER = {
  SUCCESS: "Add customer successfully",
  FAILED: "Failed to add customer",
};

export const MESSAGE_EDIT_CUSTOMER = {
  SUCCESS: "Edit customer successfully",
  FAILED: "Failed to edit customer",
};

export const MESSAGE_DELETE_CUSTOMER = {
  SUCCESS: "Delete customer successfully",
  FAILED: "Failed to delete customer",
};


export const MESSAGE_GET_CUSTOMER = {
  SUCCESS: "Get customer successfully",
  FAILED: "Failed to get customer",
};

