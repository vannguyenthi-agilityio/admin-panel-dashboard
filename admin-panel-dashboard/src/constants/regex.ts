export const REGEX = {
  EMAIL:
    /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|.(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
  NAME: /^[A-Za-z.-]+(\s*[A-Za-z.-]+)*$/,
  PHONE_NUMBER: /^\(?([0-9]{2})\)?[-. ]?([0-9]{4})[-. ]?([0-9]{4})$/,
  NUMBER: /^\d*[0-9]\d*$/
};
