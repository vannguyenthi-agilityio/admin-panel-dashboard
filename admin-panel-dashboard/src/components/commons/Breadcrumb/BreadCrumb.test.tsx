import {
  render,
} from '@testing-library/react';
import '@testing-library/jest-dom';

// Components
import BreadCrumb from '.'

// Constants
import { ICON_ITEM_BREADCRUMB } from '@/constants';

describe('BreadCrumb component', () => {
  it('Renders breadcrumb default component', () => {
    const {container} = render(<BreadCrumb />)
    expect(container).toMatchSnapshot();
  });

  it('Renders breadCrumb with Icon Item component', () => {
    const {container} = render(<BreadCrumb itemsBreadcrumb={ICON_ITEM_BREADCRUMB} />)
    expect(container).toMatchSnapshot();
  });
})
