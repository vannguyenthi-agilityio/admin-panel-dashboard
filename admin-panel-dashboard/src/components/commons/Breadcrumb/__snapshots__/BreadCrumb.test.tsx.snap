// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadCrumb component Renders breadCrumb with Icon Item component 1`] = `
<div>
  <nav
    aria-label="Breadcrumb"
  >
    <ol
      class="inline-flex items-center"
    >
      <li
        class="flex text-seldom text-sm px-0 py-2"
      >
        <a
          class="inline-flex items-center font-normal hover:text-blue-600 dark:text-gray-400 dark:hover:text-white"
          href="/"
        >
          <span
            class="mr-1"
          >
            ðŸ‘‰
          </span>
          DashBoard
        </a>
        <span
          class="mx-2 text-gray-400"
        >
          /
        </span>
      </li>
      <li
        class="flex text-tertiary text-sm px-0 py-2"
      >
        <a
          class="inline-flex items-center font-normal hover:text-blue-600 dark:text-gray-400 dark:hover:text-white"
          href="/"
        >
          <span
            class="mr-1"
          >
            ðŸ‘‰
          </span>
          Customersâ€™ List
        </a>
      </li>
    </ol>
  </nav>
</div>
`;

exports[`BreadCrumb component Renders breadcrumb default component 1`] = `
<div>
  <nav
    aria-label="Breadcrumb"
  >
    <ol
      class="inline-flex items-center"
    >
      <li
        class="flex text-seldom text-sm px-0 py-2"
      >
        <a
          class="inline-flex items-center font-normal hover:text-blue-600 dark:text-gray-400 dark:hover:text-white"
          href="/"
        >
          DashBoard
        </a>
        <span
          class="mx-2 text-gray-400"
        >
          /
        </span>
      </li>
      <li
        class="flex text-tertiary text-sm px-0 py-2"
      >
        <a
          class="inline-flex items-center font-normal hover:text-blue-600 dark:text-gray-400 dark:hover:text-white"
          href="/"
        >
          Customersâ€™ List
        </a>
      </li>
    </ol>
  </nav>
</div>
`;
